ambry-messageformat/src/main/java/com.github.ambry.messageformat/MessageFormatRecord.java

```diff
@@ -1669,7 +1710,7 @@ public static BlobData deserializeBlobRecord(CrcInputStream crcStream) throws IO
       if (dataSize > Integer.MAX_VALUE) {
         throw new IOException("We only support data of max size == MAX_INT. Error while reading blob from store");
       }
-      ByteBufferInputStream output = new ByteBufferInputStream(crcStream, (int) dataSize);
+      ByteBufferInputStream output = getByteBufferInputStreamForBlobRecord(crcStream, (int) dataSize);
```

Comment:
Why cast dataSize to int? Not possible to be long?

Reply:
updated.

Reply:
I guess I would argue that for this method, int is really the right type given that it materializes the stream contents into a single array, which has a max size of Integer.MAX_VALUE (the check which is enforced on the previous line). Functionally though, it shouldn't matter much b/c of that chunk